File: tests/ocaml/simple.ml
module Foo = struct
  type color = Red | Green | Blue

  let factorial n =
    if n = 0 then 1
    else n * factorial (n - 1)

  let msg = "hello"
  let x = 42
  let b = true
end

open Foo

let () = print_int (factorial 5)

File: tests/go/simple.go
package main

import "fmt"

type Color int

const (
	Red   Color = 0
	Green Color = 1
	Blue  Color = 2
)

func factorial(n int) int {
	if n == 0 {
		return 1
	}
	return n * factorial(n-1)
}

func main() {
	msg := "hello"
	x := 42
	fmt.Println(msg, factorial(x))
}
